storm-client/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -233,7 +235,7 @@ public static void submitTopologyAs(String name, Map<String, Object> topoConf, S
         conf.putAll(topoConf);
         topoConf.putAll(prepareZookeeperAuthentication(conf));
 
-        validateConfs(conf, topology);
+        validateConfs(conf, topology, name);
```

Comment:
I feel like this is not the right place to detect cycles.  We should probably put it in another function, like 
`StormCommon.validateBasic`

https://git.vzbuilders.com/storm/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/StormCommon.java#L156

or create a new method.

Reply:
Created new method StormCommon.validateCycleFree

Reply:
Moving it to Utils is also okay, like 
`Utils.validateTopologyBlobStoreMap`
`Utils.validateTopologyName`
etc.

Reply:
@bipinprasad   
Sorry for not being clear at the first comment. What do you think about moving it to Utils?^

Reply:
Ok. Will switch.

Reply:
Moved from StormCommons to Utils.